src/include/sof/schedule.h

```diff
@@ -25,113 +25,98 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
  */
 
-#ifndef __INCLUDE_SOF_SCHEDULE_H__
-#define __INCLUDE_SOF_SCHEDULE_H__
+/* Generic schedule api header */
+#ifndef __INCLUDE_SOF_SCHEDULER_H__
+#define __INCLUDE_SOF_SCHEDULER_H__
 
 #include <stdint.h>
-#include <stddef.h>
-#include <errno.h>
-#include <sof/sof.h>
-#include <sof/lock.h>
 #include <sof/list.h>
-#include <sof/work.h>
-#include <sof/wait.h>
 
-/* schedule tracing */
-#define trace_schedule(format, ...) \
-	trace_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define trace_schedule_error(format, ...) \
-	trace_error(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define tracev_schedule(format, ...) \
-	tracev_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
+enum {
+	SOF_EDF_TASK = 0,
+	SOF_LOW_LATENCY_TASK,
+	SOF_TASKS_COUNT
+};
 
-struct schedule_data;
-struct sof;
+#define SOF_TASK_PRI_0		0	/* priority level 0 (lowest) */
```

Comment:
Do we really need these defines for each priority?

Reply:
Now, I'm using defines only for 3 levels.